### PR TITLE
[Build] fix FreeBSD build

### PR DESCRIPTION
**Pull request by Viste**
**Description**
For now build on freebsd is broken in this PR i plan to get it works again

**Todo / Checklist**
- [x] Target is branch develop.
- [x] Detailed description about this pull request.
- [x] Checked AE-Coding standards.

**Tests Performed:** 
- [] Build AE.
- [] Server startup.
- [] Log into world.

### CODE REVIEW

**Comment by schnek** [comment]
image 🤣😁

**Comment by Viste** [comment]
> image roflgrin

Sorry, seems i gone blind.

**Comment by Zyres** [comment]
@Viste: How is it going :-)

**Comment by Viste** [comment]
> @Viste: How is it going :-)

real life troubles take some time so i don't do anything about 5 days.

Now i can start again and i have a question about 
https://github.com/AscEmu/AscEmu/blob/develop/src/logonserver/Realm/RealmManager.hpp#L33-L34
Why? Why? Is it necessary?

File: .gitignore [hunk 1]
```diff
@@ -15,3 +15,4 @@ configs/realms.conf
 configs/world.conf
 /cmake-build-debug/
 cmake-build-*
+/build*/
```
Comment (Viste) [inline .gitignore:18]: because i prefer build in src/build ;)

**Comment by Viste** [comment]
ready https://paste2.org/Ob9cZdNs

File: src/logonserver/Realm/RealmManager.hpp [hunk 2]
```diff
@@ -30,8 +30,8 @@ namespace AscEmu::Realm
         std::chrono::high_resolution_clock::time_point lastPing;
     };
 
-    class ::AuthSocket;
-    class ::LogonCommServerSocket;
+    //class ::AuthSocket;
+    //class ::LogonCommServerSocket;
```
Comment (Viste) [inline src/logonserver/Realm/RealmManager.hpp:34]: idk why this here, for fbsd and linux clang build, it makes error, so i comment this

**Comment by Viste** [comment]
also if someone have questions about PR better send it via discord :)

File: src/logonserver/Realm/RealmManager.hpp (hunk 2 above)
Comment (schnek) [reply src/logonserver/Realm/RealmManager.hpp:34]: image

File: src/world/Map/MapScriptInterface.cpp [hunk 3]
```diff
@@ -185,7 +185,7 @@ Creature* MapScriptInterface::SpawnCreature(MySQLStructure::CreatureSpawn* sp, b
     uint8 Gender = creature_properties->GetGenderAndCreateRandomDisplayID(&sp->displayid);
     Creature* p = this->mapMgr.CreateCreature(sp->entry);
     ARCEMU_ASSERT(p != NULL);
-    p->Load(sp, static_cast<unsigned int>(reinterpret_cast<uintptr_t>(NULL)), nullptr);
+    p->Load(sp, reinterpret_cast<uintptr_t>(NULL), nullptr);
```
Comment (Appled) [inline src/world/Map/MapScriptInterface.cpp:188]: You could remove all casts and replace the null with 0.

Comment (Viste) [reply src/world/Map/MapScriptInterface.cpp]: yes i know, this option i use when i surrender

**Comment by Zyres** [comment]
Give me sign if this PR is ready :-)

**Comment by Viste** [comment]
> Give me sign if this PR is ready :-)

its ready

**Pull request merged**